src/go/rpk/pkg/cli/cluster/partitions/toggle.go

```diff
@@ -0,0 +1,106 @@
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+func newPartitionEnableCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var partitions []int
+	cmd := &cobra.Command{
+		Use:   "enable [TOPIC]",
+		Short: "Enable partitions of a topic",
+		Long: `Enable partitions of a topic
+
+You may enable all partitions of a topic or you may select which partitions to
+enable with the '--partitions' flag.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = runToggle(cmd.Context(), cl, partitions, topics[0], "enable")
+			out.MaybeDieErr(err)
+		},
+	}
+	cmd.Flags().IntSliceVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions you want to enable")
+	return cmd
+}
+
+func newPartitionDisableCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var partitions []int
+	cmd := &cobra.Command{
+		Use:   "disable [TOPIC]",
+		Short: "Disable partitions of a topic",
+		Long: `Disable partitions of a topic
+
+You may disable all partitions of a topic or you may select which partitions to
+disable with the '--partitions' flag.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = runToggle(cmd.Context(), cl, partitions, topics[0], "disable")
+			out.MaybeDieErr(err)
+		},
+	}
+	cmd.Flags().IntSliceVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions you want to enable")
+	return cmd
+}
+
+func runToggle(ctx context.Context, cl *adminapi.AdminAPI, partitions []int, topic, verb string) error {
+	isDisable := verb == "disable"
+	ns, topicName := nsTopic(topic)
+	if len(partitions) == 0 { // Enable/Disable all in the topic
+		err := cl.ToggleAllTopicPartitions(ctx, isDisable, ns, topicName)
+		if err != nil {
+			return fmt.Errorf("failed to %v all partitions in topic %v/%v: %v", verb, ns, topicName, err)
+		}
+		fmt.Printf("Successfully %vd all partitions in topic %v/%v\n", verb, ns, topicName)
+		return nil
+	}
+	g, egCtx := errgroup.WithContext(ctx)
+	for _, p := range partitions {
+		p := p
+		g.Go(func() error {
+			err := cl.ToggleTopicPartitions(egCtx, isDisable, ns, topicName, p)
+			if err != nil {
+				return fmt.Errorf("failed to %v partition %v in topic %v/%v: %v", verb, p, ns, topicName, err)
+			}
+			return nil
+		})
+	}
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Successfully %vd %v partitions in the topic %v/%v\n", verb, partitions, ns, topicName)
```

Comment:
Can this output, and the other printf in this function, move to the `Run` function?